src/redux/actions/pipeline/runQC.js

```diff
@@ -42,12 +56,27 @@ const runQC = (experimentId) => async (dispatch, getState) => {
   const { processing } = getState().experimentSettings;
   const { changedQCFilters } = processing.meta;
 
-  if (changedQCFilters.size === 1 && changedQCFilters.has('configureEmbedding')) {
```

Comment:
With changedQCFilters.size === 1 removed, now that means that if there's more than one changed filter, running it with changed embedding or clustering in the last step will discard them all and run only embedding / clustering?

Reply:
`changedQCFilters.size === 1` is removed because we will now be looking for either exactly `embeddingSettings` , `clusteringSettings`, or both. If both are changed but nothing else we also need to run this conditional block

Reply:
Sorry I cant test in staging because i dont have an hms account but what i see is that now this will be run even if there are other `changedQCFilters` from previous steps. There is a `return` on line 81, so that means qc wont be triggered. 
What will happen if i change for example the mitochondrial content filter and then go to configure embedding and change clustering / embedding there? I think mitochondrial filter changes will be discarded and clustering / embedding will be ran right?

Reply:
If you send me an email I'll create an account for you to test. Here is why it doesn't run if there are other steps changes:

```javascript
const otherChanged = [...changedQCFilters].some((value) => value !== 'embeddingSettings' && value !== 'clusteringSettings');
```
This checks to see if there are other settings changed. If there **are not**, then the below conditional block runs. See added comments to hopefully clarify:

```javascript
if (!otherChanged) {

// this runs because exactly embedding or clustering or both has changed (but nothing else)
await dispatch(saveProcessingSettings(experimentId, 'configureEmbedding'));

if (embeddingChanged) {
      runOnlyConfigureEmbedding(
        experimentId,
        processing.configureEmbedding.embeddingSettings.method,
        dispatch,
      );
    }
    if (clusteringChanged) {
      runOnlyClustering(
        experimentId,
        processing.configureEmbedding.clusteringSettings.methodSettings.louvain.resolution,
        dispatch,
      );
    }
    // we return because we are in the !otherChanged block so nothing else changed and we don't re-run QC
    return;
}
 ```

Reply:
oh alright sorry

Reply:
Not at all -- thank you -- always good to have a careful review :)